src/bootstrap/test.rs

```diff
@@ -704,6 +704,39 @@ impl Step for RustdocUi {
     }
 }
 
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub struct ErrorCodesCheck {
```

Comment:
There is already a step to run rustdoc tests on the error index: https://github.com/rust-lang/rust/blob/175631311716d7dfeceec40d2587cde7142ffa8c/src/bootstrap/test.rs#L1393

Does it not work properly?

Reply:
This is not for the same purpose. `ErrorIndex` generates it, but it doesn't test that the code examples inside the explanations are compiling (or not) as expected.

Reply:
Yes it does. It uses `error_index_generator` to concatenate all of the error code docs into a single `error-index.md` file and then runs `rustdoc --test` on that: https://github.com/rust-lang/rust/blob/175631311716d7dfeceec40d2587cde7142ffa8c/src/bootstrap/test.rs#L1411-L1436

Reply:
And I can tell that it doesn't work since a lot of error codes I fixed in this PR weren't detected.

Reply:
But *why* doesn't it work? If we have existing code that overlaps this one, ideally we'd fix the pre-existing code, rather than replace it wholesale (or at least, we should make sure that we're replacing it, rather than just also adding atop it).

Reply:
This is a good question. Let me check again.